### ISSUE TITLE
[1630] Design for Center Focused Channel Vertically

### ISSUE THREAD

**Issue by athomasmoz** [mozilla-mobile/firefox-tv#1312]
### Why/User Benefit/User Problem

As a user, I want the tiles I'm looking at to be centered on the screen so it's easy to see where my remote is focused and I can see what's above & below before I click the arrows.

### What / Requirements

- The focused tile is centered between the top & bottom of the screen

![image](https://user-images.githubusercontent.com/43346208/47121359-44862680-d227-11e8-8ccf-09e5a6f10df5.png)

## Design

Source file is available on Google Drive `~/Firefox - Fire TV/Mocks/FTTV-Master.sketch`

**Comment by aminalhazwani** [comment]
List of cases when the channel is focus in the center. For example when the channel is not immediate after the toolbar.

<img width="902" alt="image" src="https://user-images.githubusercontent.com/1721875/47566549-66b92c00-d92c-11e8-850b-0087b8d617f4.png">

Or when the channel is in the split screen.

<img width="906" alt="image" src="https://user-images.githubusercontent.com/1721875/47566578-7a649280-d92c-11e8-8552-9ce2b1dc7acb.png">

Exceptions are: when channel is first after toolbar in non-split screen.

<img width="900" alt="image" src="https://user-images.githubusercontent.com/1721875/47566617-90725300-d92c-11e8-9256-be4c8ef2eb67.png">

Or when is last channel.

<img width="921" alt="image" src="https://user-images.githubusercontent.com/1721875/47566630-9c5e1500-d92c-11e8-95ad-442ffa137ef6.png">

**Comment by Baron-Severin** [comment]
Blocked on #1669 (an intermediate state broken out of this story)

**Comment by mcomella** [comment]
This seems to be a common implementation so look for existing solutions:
- The Android TV support library
- Various blog posts

**Comment by mcomella** [comment]
@brampitoyo Is this something we might want to do? If not, please close. If so, can you rename the title to [1630] and maybe resurface it when we're focusing on homescreen polish? Hopefully, it'll come up sooner in triage though.

**Comment by brampitoyo** [comment]
@mcomella Yes. This is something we want to do when we focus on homescreen polishes.

I will post the specs shortly.

**Comment by brampitoyo** [comment]
@mcomella the specs was posted in https://github.com/mozilla-mobile/firefox-tv/issues/2238#issue-442900023:

> Whenever any channel needs to be vertically centred, it will have a 116dp padding to the top of the screen.
> 
> • Pinned site [is] vertically positioned at the centre of the screen.
> • 3rd-to-last channel is still vertically centred.
> • 2rd-to-last and last channels aren’t vertically centred. They are pinned to the bottom of the screen.

**Comment by mcomella** [comment]
This issue was blocked on split overlay work that is no longer relevant ([source](https://github.com/mozilla-mobile/firefox-tv/issues/1312#issuecomment-453700817)): removing "blocked" label.

**Comment by mcomella** [comment]
@athomasmoz FYI: on the current homescreen, when navigating down from Pocket to Pinned Tiles, the screen scrolls so you can see the News & Politics section. However, when focusing the News & Politics section, the screen does not scroll so you can't see the categories below that. I wonder if this is preventing users from discovering more than News & Politics channel.

**Comment by brampitoyo** [comment]
@athomasmoz This issue should be solved with the behaviour I’ve described above. Here are some illustrations to help clarify it.

1. The toolbar and first channel don’t need to be vertically centred.

![](https://user-images.githubusercontent.com/51007/58397369-c3d4b600-80a4-11e9-8208-fc297f3729ba.png)
![](https://user-images.githubusercontent.com/51007/58397376-c8996a00-80a4-11e9-9231-bb470c6293c2.png)

2. Most other channels need to be centred.

![](https://user-images.githubusercontent.com/51007/57743144-0e0a7e80-7718-11e9-9369-5d4deefdc7b6.png)
![](https://user-images.githubusercontent.com/51007/57743418-1911de80-7719-11e9-9ed5-5ecaa55f8367.png)
![](https://user-images.githubusercontent.com/51007/57744190-4f049200-771c-11e9-8eca-7b99d2602391.png)

3. second-to-last and last channel aren’t vertically centred. They’re pinned to the bottom of the screen.

![](https://user-images.githubusercontent.com/51007/57744215-6fcce780-771c-11e9-8680-2186b3e5ca0e.png)
![](https://user-images.githubusercontent.com/51007/57744216-722f4180-771c-11e9-9d15-9bad9ed07c90.png)

**Comment by athomasmoz** [comment]
UX completed. Implementation ticket #2573
